docs/places.mdown

```diff
@@ -0,0 +1,107 @@
+Rust Places
+===========
+
+Places is a library for storing and syncing bookmarks and history. It exposes high-level APIs for common use cases, like recording history visits, autocompleting visited URLs, clearing and expiring history, and managing bookmarks. It also provides engines for syncing visited pages and bookmarks through Firefox Sync.
+
+In the past, all Mozilla browsers handled history and bookmark storage differently. Desktop used its [own implementation, written in a mix of C++ and JavaScript](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/Places); Firefox for Android had [one written in Java](https://searchfox.org/mozilla-central/rev/f1c7ba91fad60bfea184006f3728dd6ac48c8e56/mobile/android/base/java/org/mozilla/gecko/db/BrowserDB.java); and Firefox for iOS had a [third implementation in Swift](https://github.com/mozilla-mobile/firefox-ios/blob/ceb3cc9f0cc1ad8a90ad465b8c74a855121f319d/Storage/SQL/BrowserSchema.swift). All platforms used [SQLite](https://sqlite.org/) for storage, with a similarly-shaped schema, but the similarities ended there. Each implementation evolved independently, dictated by immediate product needs, and concepts like [frecency](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/Places/Frecency_algorithm) and address autocomplete heuristics weren't shared at all. Each platform also had its own sync implementation. The goal of Rust Places is to unify these features across all platforms.
```

Comment:
µnit: We tend to keep our lines no longer than 100-120 chars or so. I don't know the actual number (and it's a soft limit, not a hard one), but take a look in the howtos for other examples.